src/bundle/Resources/public/admin/js/ibexa-automated-translation.js

```diff
@@ -1,57 +1,61 @@
+            target.classList.add('is-checked');
+            input.setAttribute('checked', true);
         }
-        return false;
-    });
+    };
 
-    $container.find(".ibexa-data-source__label.is-checked").each(() => {
-        $form.find('.ibexa-btn--create-translation').prop("disabled", false);
-    });
-
-    $container.find(".ibexa-dropdown__source select option:selected").each(() => {
-        $form.find('.ibexa-btn--create-translation').prop("disabled", false);
-    });
-
-    $("form[name=add-translation]").submit(function () {
-        let targetLang = $("select[name=add-translation\\[language\\]]").val();
-        let sourceLang = $("select[name=add-translation\\[base_language\\]]").val();
-        let mapping = $container.data('languages-mapping');
-        let $serviceSelector = $("#add-translation_translatorAlias");
-        let serviceAlias = $serviceSelector.val();
-        if ($serviceSelector.is("[type=checkbox]") && !$serviceSelector.is(":checked")) {
-            serviceAlias = '';
-        }
+    const submitForm = () => {
+        const targetLang = form.querySelector('select[name="add-translation[base_language]"]').value;
+        const sourceLang = form.querySelector('select[name="add-translation[language]"]').value;
+        const { languagesMapping } = container.dataset;
+        const serviceSelector = form.querySelector('#add-translation_translatorAlias');
+        const serviceAlias = serviceSelector.value;
 
-        if (!serviceAlias.length) {
+        if ((serviceSelector.type === 'checkbox' && !serviceSelector.checked) || !serviceAlias.length) {
             return true;
         }
 
-        let translationAvailable = (typeof sourceLang === 'undefined' || -1 !== $.inArray(sourceLang, mapping[serviceAlias])) && (-1 !== $.inArray(targetLang, mapping[serviceAlias]));
-        if (false === translationAvailable) {
-            $error.removeClass("invisible");
-            if ($container.find(".ibexa-field-edit--ezboolean .ibexa-data-source__label").hasClass('is-checked')) {
-                $container.find(".ibexa-field-edit--ezboolean .ibexa-data-source__label").click();
+        const translationAvailable =
+            (typeof sourceLang === 'undefined' || sourceLang.indexOf(languagesMapping[serviceAlias]) !== -1) &&
+            targetLang.indexOf(targetLang, languagesMapping[serviceAlias]) !== -1;
+
+        if (!translationAvailable) {
+            error?.classList.remove('invisible');
+
+            if (sourceLabel.classList.contains('is-checked')) {
+                sourceLabel.click();
+
                 return false;
             }
         }
+
         return true;
+    };
+
+    form.addEventListener('click', () => error?.classList.add('invisible'));
+    sourceLabel?.addEventListener('click', toggleClass);
+
+    container.querySelectorAll('.ibexa-data-source__label.is-checked').forEach(() => {
+        form.querySelector('.ibexa-btn--create-translation').removeAttribute('disabled');
+    });
```

Comment:
I think there's something missing in here, as you iterate by `.ibexa-data-source__label.is-checked`, but inside you don't use anything from this iterable array?
I guess it was supposed to remove disabled from button if sth is checked, but in such case it should be enough:
```suggestion
    if (container.querySelector('.ibexa-data-source__label.is-checked')) {
        form.querySelector('.ibexa-btn--create-translation').removeAttribute('disabled');
    };
```